x/ratelimit/keeper/grpc_query_test.go

```diff
@@ -0,0 +1,303 @@
+				s.Require().Equal(sampleRateLimitA.MaxPercentSend, resp.RateLimits[0].Quota.MaxPercentSend)
+				s.Require().Equal(sampleRateLimitA.MaxPercentRecv, resp.RateLimits[0].Quota.MaxPercentRecv)
+				s.Require().Equal(sampleRateLimitA.MinRateLimitAmount, resp.RateLimits[0].MinRateLimitAmount)
+				s.Require().Equal(sampleRateLimitA.DurationHours, resp.RateLimits[0].Quota.DurationHours)
+				s.Require().Equal(sampleRateLimitB.Denom, resp.RateLimits[1].Path.Denom)
+				s.Require().Equal(sampleRateLimitB.ChannelID, resp.RateLimits[1].Path.ChannelID)
+				s.Require().Equal(sampleRateLimitB.MaxPercentSend, resp.RateLimits[1].Quota.MaxPercentSend)
+				s.Require().Equal(sampleRateLimitB.MaxPercentRecv, resp.RateLimits[1].Quota.MaxPercentRecv)
+				s.Require().Equal(sampleRateLimitB.MinRateLimitAmount, resp.RateLimits[1].MinRateLimitAmount)
+				s.Require().Equal(sampleRateLimitB.DurationHours, resp.RateLimits[1].Quota.DurationHours)
+			},
+		},
+		{
+			"query by chain id that does not exist",
+			&types.QueryRateLimitsByChannelIDRequest{
+				ChannelID: "channel-10",
+			},
+			false,
+			func(resp *types.QueryRateLimitsByChannelIDResponse) {
+				s.Require().Empty(resp.RateLimits)
+			},
+		},
+		{
+			"query by invalid chain id",
+			&types.QueryRateLimitsByChannelIDRequest{
+				ChannelID: "invalid/ChannelID",
+			},
+			true,
+			nil,
+		},
+	} {
+		s.Run(tc.name, func() {
+			resp, err := s.querier.RateLimitsByChannelID(sdk.WrapSDKContext(s.ctx), tc.req)
+			if tc.expectErr {
+				s.Require().Error(err)
+			} else {
+				s.Require().NoError(err)
+				tc.postRun(resp)
+			}
+		})
+	}
+}
+
+func (s *KeeperTestSuite) TestAllWhitelistedAddresses() {
+	// Add some sample whitelisted addresses
+	whitelistedAddrPairs := []types.WhitelistedAddressPair{
+		{Sender: s.addr(1).String(), Receiver: s.addr(2).String()},
+		{Sender: s.addr(3).String(), Receiver: s.addr(4).String()},
+		{Sender: s.addr(5).String(), Receiver: s.addr(6).String()},
+	}
+
+	for _, wap := range whitelistedAddrPairs {
+		s.keeper.SetWhitelistedAddressPair(s.ctx, wap)
+	}
+
+	for _, tc := range []struct {
+		name      string
+		req       *types.QueryAllWhitelistedAddressesRequest
+		expectErr bool
+		postRun   func(*types.QueryAllWhitelistedAddressesResponse)
+	}{
+		{
+			"nil request",
+			nil,
+			true,
+			nil,
+		},
+		{
+			"happy case",
+			&types.QueryAllWhitelistedAddressesRequest{},
+			false,
+			func(resp *types.QueryAllWhitelistedAddressesResponse) {
+				s.Require().Len(resp.GetAddressPairs(), 3)
+			},
+		},
+	} {
+		s.Run(tc.name, func() {
+			resp, err := s.querier.AllWhitelistedAddresses(sdk.WrapSDKContext(s.ctx), tc.req)
+			if tc.expectErr {
+				s.Require().Error(err)
+			} else {
+				s.Require().NoError(err)
+				tc.postRun(resp)
+			}
+		})
+	}
+}
+
+func (s *KeeperTestSuite) TestChanOpenInit() {
+	// Create client and connections on both chains
+	path := ibctesting.NewPath(s.chainA, s.chainB)
+	s.coordinator.SetupConnections(path)
```

Comment:
I think that this current `path` doesn't set portID and version which is used to retrieve capability